### PR TITLE
Create add-item article

### PR DESCRIPTION
**Pull request by MrEssCee**
Original article for adding an item to a cart when using Umbraco Commerce

## Description

How to add an item to the Umbraco Commerce cart

_What did you add/update/change?_

## Type of suggestion

* [ ] Typo/grammar fix
* [ ] Updated outdated content
* [X] New content
* [ ] Updates related to a new version
* [ ] Other

## Product & version (if relevant)
Umbraco 13

## Deadline (if relevant)

_When should the content be published?_
ASAP

### CODE REVIEW

**Comment by sofietoft** [comment]
Thanks for the PR @MrEssCee ! 💪 

We'll make sure to get this reviewed as soon as possible.

File: 13/umbraco-commerce/how-to-guides/add-item [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+- The Product is returned as a ProductSnapshot which is Umbraco Commerce obtaining the page ID and carrying out necessary processes to bring in the data which it can use for further processing.
+- Finally we need to calculate the price which is then displayed without VAT (although can be also displayed with VAT)
+
+To display this we need to add some markup or at least amend it to include a button to add an item. In the same file add the below
+
+```csharp
+@using (Html.BeginUmbracoForm("AddToCart", "CartSurface"))
+{
+	@Html.Hidden("productReference", Model.Key.ToString())
+	<h1>@Model.ProductTitle</h1>
```
Comment (umbracotrd) [inline 13/umbraco-commerce/how-to-guides/add-item:31]:
```suggestion
	<h1>@Model.Value<string>("productTitle")</h1>
```

File: 13/umbraco-commerce/how-to-guides/add-item [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+	<h2>@Model.ProductDescription</h2>
```
Comment (umbracotrd) [inline 13/umbraco-commerce/how-to-guides/add-item:32]:
```suggestion
	<h2>@Model.Value<string>("productDescription")</h2>
```

File: 13/umbraco-commerce/how-to-guides/add-item [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+## Add a product to the Cart
+You will need the Frontend to be setup to add allow an item to be added to the cart. This can be done by adding a button to the frontend that will call the Action to add the item to the cart.
+
+Create a new View, we will call it product.cshtml with the below code
+
+```csharp
+@{
+var store = Model.Value<StoreReadOnly>("store", fallback: Fallback.ToAncestors);
+var product = CommerceApi.Instance.GetProduct(store.Id, Model.Key.ToString(), "en-GB");
+var price = product.TryCalculatePrice();
```
Comment (umbracotrd) [inline 13/umbraco-commerce/how-to-guides/add-item:16]:
```suggestion
var price = product.TryCalculatePrice().ResultOrThrow("Unable to calculate product price");
```

File: 13/umbraco-commerce/how-to-guides/add-item [hunk 4]
```diff
@@ -0,0 +1,182 @@
...
+To display this we need to add some markup or at least amend it to include a button to add an item. In the same file add the below
+
+```csharp
+@using (Html.BeginUmbracoForm("AddToCart", "CartSurface"))
+{
+	@Html.Hidden("productReference", Model.Key.ToString())
+	<h1>@Model.ProductTitle</h1>
+	<h2>@Model.ProductDescription</h2>
+
+	<p>Our price excluding VAT <strong>@price.Result?.WithoutTax.ToString("C0") </strong></p>
```
Comment (umbracotrd) [inline 13/umbraco-commerce/how-to-guides/add-item:34]:
```suggestion
	<p>Our price excluding VAT <strong>@price.WithoutTax.ToString("C0") </strong></p>
```

File: 13/umbraco-commerce/how-to-guides/add-item [hunk 5]
```diff
@@ -0,0 +1,182 @@
...
+		<button type="submit">Add to Basket</button>
+	}
+
+    @await Html.PartialAsync("...../Feedback.cshtml")
+}
+```
+
+The hidden field is using the productReference to be passed across to the Controller.
+
+{% hint style="warning" %}
```
Comment (umbracotrd) [inline 13/umbraco-commerce/how-to-guides/add-item:51]: Remove this hint

**Review by umbracotrd — COMMENTED** [review]
@sofietoft : Fix syntax in code example

File: 13/umbraco-commerce/how-to-guides/add-item (hunk 5 above)
Comment (MrEssCee) [reply 13/umbraco-commerce/how-to-guides/add-item]: Removed

File: 13/umbraco-commerce/how-to-guides/add-item [hunk 6]
```diff
@@ -0,0 +1,192 @@
...
+description: How-To Guide to add an item to your cart.
+---
+
+# Add item to Cart
+
+To add an item to the cart, you need to set up Umbraco with a store and add the relevant properties to allow the store to interact with Umbraco. Learn more by following the [Getting started with Umbraco Commerce: The Backoffice tutorial](../tutorials/getting-started-with-commerce).
+
+You will need the front end to be set up to allow an item to be added to the cart. This can be done by adding a button to the front end to call the Action to add the item to the cart.
+
+Create a new Document Type with the template. Call it **Product Page** with the following property aliases: `productTitle`, `productDescription`, `price`, `stock`.
```
Comment (sofietoft) [inline 13/umbraco-commerce/how-to-guides/add-item:11]: Which Property Editors should be used for these properties?

**Review by sofietoft — CHANGES_REQUESTED** [review]
Great guide! 💪 
I've taken the liberty to correct some grammar and apply our style guide to the article.

Also, could you look into adding the code snippets to code blocks, using the following docs: https://docs.gitbook.com/content-editor/blocks/code-block#representation-in-markdown 
Don't mind about the line numbers and the wrap, but adding the file name would be great 🙌 

Also, I've asked a question regarding the properties on the Document Type.

Hope it all makes sense!

**Comment by sofietoft** [comment]
Also realized that the file name here is a bit wrong.
It needs to have the .md file extension to work with our docs 🙌

**Comment by sofietoft** [comment]
Could you also add the article to the SUMMARY file?
Go up in the file structure and file the closest SUMMARY.md file - that will be for this version of the Commerce docs.
Add a link to this article, using the title of the article as the link title 💪 

If it's not added to the SUMMARY file, it will not be part of the published docs.
Happy to do this as well - just let me know.

File: 13/umbraco-commerce/how-to-guides/add-item (hunk 6 above)
Comment (MrEssCee) [reply 13/umbraco-commerce/how-to-guides/add-item]: @sofietoft Ive added those, sorry not sure if the correct styling was applied as for me the Preview feature doesnt seem to be working :-(

**Comment by MrEssCee** [comment]
> Also realized that the file name here is a bit wrong. It needs to have the .md file extension to work with our docs 🙌

renamed 👍

File: 13/umbraco-commerce/how-to-guides/add-item (hunk 6 above)
Comment (umbracotrd) [reply 13/umbraco-commerce/how-to-guides/add-item]: - `productTitle`: text string
- `productDescription`: text string
- `price`: Umbraco Commerce: Price
- `stock`: Umbraco Commerce: Stock

**Comment by sofietoft** [comment]
Thanks for applying the suggestions @MrEssCee ! 💪 

I've taken the liberty to add the article in this PR to the SUMMARY file for 13/commerce.
Now we can also preview the article in the docs - use the GitBook link with docs.umbraco.com in the Checks window below.

Are you up for adding the other articles to the SUMMARY as well?
It would be best if you can do it in the same PR as the article - that will let us use the preview 😁

**Review by sofietoft — APPROVED** [review]

**Comment by MrEssCee** [comment]
> Thanks for applying the suggestions @MrEssCee ! 💪
> 
> I've taken the liberty to add the article in this PR to the SUMMARY file for 13/commerce. Now we can also preview the article in the docs - use the GitBook link with docs.umbraco.com in the Checks window below.
> 
> Are you up for adding the other articles to the SUMMARY as well? It would be best if you can do it in the same PR as the article - that will let us use the preview 😁

@sofietoft I think i did this in 

> Thanks for applying the suggestions @MrEssCee ! 💪
> 
> I've taken the liberty to add the article in this PR to the SUMMARY file for 13/commerce. Now we can also preview the article in the docs - use the GitBook link with docs.umbraco.com in the Checks window below.
> 
> Are you up for adding the other articles to the SUMMARY as well? It would be best if you can do it in the same PR as the article - that will let us use the preview 😁

Completed 👍

**Comment by sofietoft** [comment]
Looks great @MrEssCee ! 💪 

We'll not be able to preview the other articles you've added, but it looks like all checks out in the Markdown department.
Once the PR with the delete article gives all green lights on the checks I'll start merging these in 💪 

Thanks again for all the word you've put in here.
It will be SO great to finally have some more Commerce guide in our docs 😄 🎉

**Pull request merged**